Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,45 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python2?
+        requirements.any? do |req|
+          req.is_a?(PythonRequirement) && !req.is_a?(Python3Requirement) &&
```

Comment:
I'm suggesting a method that is invoked as either `needs_python?("python")` or `needs_python?("python3")`. This would also help simplify the logic in `virtualenv_install_with_resources`.